archetype/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,97 @@
+import java.nio.file.Files
+import org.apache.commons.io.FileUtils
+
+def jakartaVersion = request.properties["jakartaVersion"].trim()
+def docker = request.properties["docker"].trim()
+def outputDirectory = new File(request.getOutputDirectory(), request.getArtifactId())
+
+validateInput(jakartaVersion, outputDirectory);
+
+switch (request.properties["runtime"]) {
```

Comment:
I recommend adding a bit more structure to the code. This switch block could be moved to a separate method, e.g. `generateRuntime()`

Reply:
Good point. Done!